EIPS/eip-7775.md

```diff
@@ -0,0 +1,102 @@
+discussions-to: https://ethereum-magicians.org/t/eip-7775-burn-opcode/21287
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-09-30
+---
+
+## Abstract
+
+This proposal introduces a new `BURN` opcode, designed to burn native tokens at the given address.
+
+## Motivation
+
+The motivation for this proposal is to provide a standardized and efficient way to burn native tokens directly within the EVM. Historically, contracts such as the BeaconDepositContract have "burned" ether by making it irrecoverable from the given address. This approach can lead to confusion and potential misuse. By introducing a dedicated `BURN` opcode, we can ensure a clear and consistent method for burning native tokens. This could become useful for Ethereum L2s when transferring ether back to the L1, as well as other EVM L1 chains that could leverage this for their cryptoeconomics. Additionally, it opens up an interesting design space for [EIP-7685](./eip-7685.md) use cases, providing a more robust and standardized mechanism for token management.
+
+## Specification
+
+The `BURN` opcode `(0xFC)` is introduced with the following behavior:
+
+1. Pops one 32-byte word from the top of the stack, treating these bytes as the `uint256` amount of native token to be burned.
+2. Retrieves the current address from the EVM execution context.
+3. Checks the balance of the current address.
+4. If the amount to be burned is greater than the balance of the current address, the opcode `MUST` revert.
+5. If the amount to be burned is 0, the execution `MUST NOT` revert.
+6. Subtracts the amount from the current address's native token balance.
+
+When `BURN` is used in the context of a DELEGATECALL or CALLCODE, the contract whose balance is to be manipulated is the contract that issued the DELEGATECALL or CALLCODE instruction.
+
+When `BURN` is used in the context of a STATICCALL the call `MUST` revert.
+
+Example pseudocode for the `BURN` opcode:
+
+```python
+def op_burn(pc, interpreter, scope):
+    # Pop the value to be burned from the stack
+    value_to_burn = scope.stack.pop()
+    
+    # Retrieve the current address from the EVM execution context
+    current_address = scope.contract.address()
+    
+    # Check the balance of the current address
+    balance = interpreter.evm.state_db.get_balance(current_address)
+    
+    # If the value to be burned is greater than the balance, revert
+    if value_to_burn > balance:
+        return "ErrInsufficientBalance"
+    
+    # If the value to be burned is 0, do not revert
+    if value_to_burn == 0:
+        return None
+    
+    # Subtract the value from the current address's balance
+    interpreter.evm.state_db.sub_balance(current_address, value_to_burn)
+    return None
+```
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
```

Comment:
For clarity, this sentence should be moved above the first uses of any of these keywords.